main.c

```diff
@@ -120,11 +120,14 @@ int main(int argc, char **argv) {
 		fprintf(stderr, "Unable to open %s\n", context.rom_file);
 		exit(1);
 	}
-
 	while (context.length--) {
 		fputc(0xff, context.rom);
 	}
-
+	if (context.input == NULL) {
+		fclose(context.rom);
+		return 0;
+	}
+	
 	input = context.input;
 	while (1) {
```

Comment:
Instead of the previous if you added, you could do a while (input) here, and even remove the if/break inside the loop.